mltsp/featurize.py

```diff
@@ -9,354 +5,212 @@
-                            else:
-                                line.append(str(obj[feat]))
-                            if feat in features_to_plot and numobjs < 300:
-                                if isinstance(obj[feat], (str, type(u''))) and \
-                                   obj[feat] not in ["None", u"None"]:
-                                    line2.append(str(obj[feat]))
-                                elif (obj[feat] is None
-                                      or obj[feat] in ["None", u"None"]):
-                                    line2.append(str(0.0))
-                                else:
-                                    line2.append(str(obj[feat]))
-                        except KeyError:
-                            print(feat, "NOT IN DICT KEYS!!!! SKIPPING...")
-                f.write(','.join(line) + '\n')
-                if numobjs < 300:
-                    f2.write(','.join(line2) + '\n')
-                if "." in str(obj["target"]) and \
-                   str(obj["target"]).replace(".", "").isdigit():
-                    obj["target"] = float(obj["target"])
-                targets.append(obj['target'])
-                num_used[str(obj['target'])] += 1
-            else:
-                cv_objs.append(obj)
-                num_held_back[str(obj['target'])] += 1
-            numobjs += 1
-    np.save(os.path.join(
-        ("/tmp" if in_docker_container else cfg.FEATURES_FOLDER),
-        "%s_targets.npy" % featureset_id), targets)
-    print("Done.")
-
-
-def featurize(
-        headerfile_path, zipfile_path, features_to_use=[],
-        featureset_id="unknown", is_test=False, already_featurized=False,
-        custom_script_path=None, in_docker_container=False):
-    """Generate features for labeled time series data.
-
-    Features are saved to the file given by
-    ``"%s_features.csv" % featureset_id``
-    and a list of corresponding targets is saved to the file given by
-    ``"%s_targets.npy" % featureset_id``
-    in the directory `cfg.FEATURES_FOLDER` (or is later copied there if
-    generated inside a Docker container).
+        fname_targets = None
+    if metadata is not None:
+        fname_metadata = metadata.loc[list(fnames)]
+    else:
+        fname_metadata = None, None
+    featureset = ft.assemble_featureset(feature_dicts, fname_targets,
+                                        fname_metadata, fnames)
+
+    if featureset_id:
+        write_features_to_disk(featureset, featureset_id)
+#    if not in_docker_container:
+#        os.remove(header_path)
+#        os.remove(data_path)
+    return featureset
+
+
+def featurize_time_series(times, values, errors=None, features_to_use=[],
+                          targets=None, meta_features=None,
+                          custom_script_path=None, labels=None):
+    """Versatile feature generation function for one or more time series
+
+    Time series data can be provided either as single arrays or as lists of
```

Comment:
We may want to expand this description (or write it up as a section in the docs), since it is so fundamental.